### ISSUE TITLE
[no-duplicate-case] Duplicate `default`s aren't reported.

### ISSUE THREAD

**Issue by manuth** [eslint/eslint#13188]
**Tell us about your environment**

```
Environment Info:

Node version: v12.4.0
npm version: v6.9.0
Local ESLint version: v6.8.0
Global ESLint version: v6.8.0 (Currently used)
```

**What parser (default, Babel-ESLint, etc.) are you using?**
```
@typescript-eslint/parser
```

**Please show your full configuration:**
```
{
  rules: {
    "no-duplicate-case": "error"
  }
}
```

<details>
<summary>Configuration</summary>

```js
{
  "env": {},
  "globals": {},
  "parser": "E:\\Users\\Manuel\\Documents\\Projekte\\Node-Modules\\ESLintPresets\\node_modules\\@typescript-eslint\\parser\\dist\\parser.js",
  "parserOptions": {
    "project": "tsconfig.json",
    "sourceType": "module"
  },
  "plugins": [
    "@typescript-eslint",
    "prefer-arrow",
    "jsdoc",
    "import",
    "@typescript-eslint/tslint"
  ],
  "rules": {
    "@typescript-eslint/explicit-member-accessibility": [
      "warn",
      {
        "accessibility": "explicit"
      }
    ],
    "@typescript-eslint/explicit-function-return-type": [
      "warn",
      {
        "allowExpressions": true
      }
    ],
    "jsdoc/no-types": [
      "warn",
      {
        "contexts": [
          "any"
        ]
      }
    ],
    "jsdoc/require-param-type": [
      "off",
      {
        "contexts": [
          "any"
        ]
      }
    ],
    "jsdoc/require-property-type": [
      "off"
    ],
    "jsdoc/require-returns-type": [
      "off",
      {
        "contexts": [
          "any"
        ]
      }
    ],
    "getter-return": [
      "off"
    ],
    "no-dupe-args": [
      "off"
    ],
    "no-dupe-keys": [
      "off"
    ],
    "no-unreachable": [
      "off"
    ],
    "valid-typeof": [
      "off"
    ],
    "no-const-assign": [
      "off"
    ],
    "no-new-symbol": [
      "off"
    ],
    "no-this-before-super": [
      "off"
    ],
    "no-undef": [
      "off"
    ],
    "no-dupe-class-members": [
      "off"
    ],
    "no-redeclare": [
      "off"
    ],
    "@typescript-eslint/adjacent-overload-signatures": [
      "warn"
    ],
    "@typescript-eslint/array-type": [
      "warn",
      {
        "default": "array-simple"
      }
    ],
    "@typescript-eslint/ban-ts-ignore": [
      "off"
    ],
    "@typescript-eslint/ban-types": [
      "error"
    ],
    "@typescript-eslint/camelcase": [
      "off"
    ],
    "@typescript-eslint/class-name-casing": [
      "off"
    ],
    "@typescript-eslint/comma-spacing": [
      "warn"
    ],
    "@typescript-eslint/consistent-type-assertions": [
      "warn"
    ],
    "@typescript-eslint/consistent-type-definitions": [
      "warn",
      "interface"
    ],
    "@typescript-eslint/default-param-last": [
      "error"
    ],
    "@typescript-eslint/func-call-spacing": [
      "warn"
    ],
    "@typescript-eslint/interface-name-prefix": [
      "off"
    ],
    "@typescript-eslint/member-delimiter-style": [
      "warn",
      {
        "singleline": {
          "delimiter": "comma",
          "requireLast": false
        },
        "multiline": {
          "delimiter": "semi",
          "requireLast": true
        }
      }
    ],
    "@typescript-eslint/member-ordering": [
      "warn",
      {
        "default": [
          "public-static-field",
          "protected-static-field",
          "private-static-field",
          "public-instance-field",
          "protected-instance-field",
          "private-instance-field",
          "public-abstract-field",
          "protected-abstract-field",
          "private-abstract-field",
          "public-field",
          "protected-field",
          "private-field",
          "static-field",
          "instance-field",
          "abstract-field",
          "field",
          "public-constructor",
          "protected-constructor",
          "private-constructor",
          "constructor",
          "signature",
          "public-static-method",
          "protected-static-method",
          "private-static-method",
          "public-instance-method",
          "protected-instance-method",
          "private-instance-method",
          "public-abstract-method",
          "protected-abstract-method",
          "private-abstract-method",
          "public-method",
          "protected-method",
          "private-method",
          "static-method",
          "instance-method",
          "abstract-method",
          "method"
        ]
      }
    ],
    "@typescript-eslint/naming-convention": [
      "warn",
      {
        "selector": "enumMember",
        "format": [
          "PascalCase"
        ]
      },
      {
        "selector": "typeLike",
        "format": [
          "PascalCase"
        ]
      },
      {
        "selector": "interface",
        "prefix": [
          "I"
        ],
        "format": [
          "PascalCase"
        ]
      },
      {
        "selector": "typeParameter",
        "prefix": [
          "T"
        ],
        "format": [
          "PascalCase"
        ]
      }
    ],
    "@typescript-eslint/no-dupe-class-members": [
      "error"
    ],
    "@typescript-eslint/no-dynamic-delete": [
      "error"
    ],
    "@typescript-eslint/no-empty-function": [
      "off"
    ],
    "@typescript-eslint/no-empty-interface": [
      "off"
    ],
    "@typescript-eslint/no-explicit-any": [
      "off"
    ],
    "@typescript-eslint/no-for-in-array": [
      "error"
    ],
    "@typescript-eslint/no-inferrable-types": [
      "error"
    ],
    "@typescript-eslint/no-misused-new": [
      "error"
    ],
    "@typescript-eslint/no-namespace": [
      "warn",
      {
        "allowDeclarations": true,
        "allowDefinitionFiles": true
      }
    ],
    "@typescript-eslint/no-parameter-properties": [
      "error"
    ],
    "@typescript-eslint/no-this-alias": [
      "off"
    ],
    "@typescript-eslint/no-unused-expressions": [
      "warn"
    ],
    "@typescript-eslint/no-unused-vars": [
      "warn"
    ],
    "@typescript-eslint/no-use-before-define": [
      "warn",
      {
        "functions": false,
        "classes": false,
        "variables": true
      }
    ],
    "@typescript-eslint/no-var-requires": [
      "warn"
    ],
    "@typescript-eslint/prefer-as-const": [
      "warn"
    ],
    "@typescript-eslint/prefer-for-of": [
      "warn"
    ],
    "@typescript-eslint/prefer-function-type": [
      "warn"
    ],
    "@typescript-eslint/prefer-namespace-keyword": [
      "error"
    ],
    "@typescript-eslint/prefer-optional-chain": [
      "warn"
    ],
    "@typescript-eslint/quotes": [
      "warn",
      "double",
      {
        "avoidEscape": true
      }
    ],
    "@typescript-eslint/semi": [
      "error",
      "always"
    ],
    "@typescript-eslint/space-before-function-paren": [
      "warn",
      {
        "anonymous": "never",
        "asyncArrow": "always",
        "named": "never"
      }
    ],
    "@typescript-eslint/triple-slash-reference": [
      "off"
    ],
    "@typescript-eslint/type-annotation-spacing": [
      "warn"
    ],
    "@typescript-eslint/unified-signatures": [
      "off"
    ],
    "array-bracket-newline": [
      "warn",
      "consistent"
    ],
    "array-bracket-spacing": [
      "warn"
    ],
    "array-element-newline": [
      "warn",
      "consistent"
    ],
    "arrow-parens": [
      "off",
      "as-needed"
    ],
    "arrow-spacing": [
      "warn"
    ],
    "block-spacing": [
      "warn"
    ],
    "brace-style": [
      "warn",
      "allman",
      {
        "allowSingleLine": true
      }
    ],
    "comma-dangle": [
      "error"
    ],
    "comma-style": [
      "warn"
    ],
    "complexity": [
      "off"
    ],
    "computed-property-spacing": [
      "warn"
    ],
    "constructor-super": [
      "error"
    ],
    "curly": [
      "off"
    ],
    "dot-notation": [
      "off"
    ],
    "eol-last": [
      "warn"
    ],
    "eqeqeq": [
      "warn",
      "always"
    ],
    "function-call-argument-newline": [
      "warn",
      "consistent"
    ],
    "generator-star-spacing": [
      "warn"
    ],
    "grouped-accessor-pairs": [
      "warn"
    ],
    "guard-for-in": [
      "off"
    ],
    "import/no-default-export": [
      "warn"
    ],
    "import/no-duplicates": [
      "warn"
    ],
    "import/order": [
      "warn",
      {
        "groups": [
          "builtin",
          [
            "external",
            "internal"
          ],
          [
            "parent",
            "sibling",
            "index"
          ]
        ],
        "pathGroups": [
          {
            "pattern": "child_process",
            "group": "builtin"
          }
        ],
        "alphabetize": {
          "order": "asc",
          "caseInsensitive": false
        }
      }
    ],
    "jsdoc/check-alignment": [
      "warn"
    ],
    "jsdoc/check-indentation": [
      "warn"
    ],
    "jsdoc/check-param-names": [
      "warn"
    ],
    "jsdoc/check-syntax": [
      "warn"
    ],
    "jsdoc/check-tag-names": [
      "warn"
    ],
    "jsdoc/check-values": [
      "warn"
    ],
    "jsdoc/empty-tags": [
      "warn"
    ],
    "jsdoc/newline-after-description": [
      "warn"
    ],
    "jsdoc/require-description": [
      "warn",
      {
        "contexts": [
          "ClassDeclaration",
          "ClassExpression",
          "ArrowFunctionExpression",
          "FunctionDeclaration",
          "FunctionExpression",
          "MethodDefinition",
          "TSEnumDeclaration",
          "TSEnumMember",
          "TSInterfaceDeclaration",
          "ClassProperty",
          "TSPropertySignature",
          "TSAbstractMethodDefinition",
          "TSCallSignatureDeclaration",
          "TSConstructSignatureDeclaration",
          "TSMethodSignature",
          "TSTypeAliasDeclaration"
        ]
      }
    ],
    "jsdoc/require-jsdoc": [
      "warn",
      {
        "require": {
          "ClassDeclaration": true,
          "ClassExpression": true,
          "ArrowFunctionExpression": true,
          "FunctionDeclaration": true,
          "FunctionExpression": true,
          "MethodDefinition": true
        },
        "contexts": [
          "TSEnumDeclaration",
          "TSEnumMember",
          "TSInterfaceDeclaration",
          "ClassProperty",
          "TSTypeAliasDeclaration",
          "TSPropertySignature",
          "TSAbstractMethodDefinition",
          "TSCallSignatureDeclaration",
          "TSConstructSignatureDeclaration",
          "TSMethodSignature",
          "TSDeclareFunction"
        ],
        "exemptEmptyFunctions": false
      }
    ],
    "jsdoc/require-param-description": [
      "warn",
      {
        "contexts": [
          "any"
        ]
      }
    ],
    "jsdoc/require-param-name": [
      "warn",
      {
        "contexts": [
          "any"
        ]
      }
    ],
    "jsdoc/require-param": [
      "warn",
      {
        "exemptedBy": [],
        "contexts": [
          "ArrowFunctionExpression",
          "FunctionDeclaration",
          "FunctionExpression",
          "TSMethodSignature",
          "TSConstructSignatureDeclaration",
          "TSCallSignatureDeclaration",
          "TSDeclareFunction",
          "TSFunctionType"
        ]
      }
    ],
    "jsdoc/require-returns": [
      "warn",
      {
        "checkGetters": false,
        "exemptedBy": [],
        "forceRequireReturn": false,
        "forceReturnsWithAsync": false
      }
    ],
    "lines-between-class-members": [
      "warn"
    ],
    "max-classes-per-file": [
      "off"
    ],
    "max-len": [
      "off"
    ],
    "multiline-ternary": [
      "warn",
      "always-multiline"
    ],
    "new-parens": [
      "warn"
    ],
    "no-async-promise-executor": [
      "off"
    ],
    "no-bitwise": [
      "off"
    ],
    "no-caller": [
      "error"
    ],
    "no-case-declarations": [
      "off"
    ],
    "no-cond-assign": [
      "error"
    ],
    "no-console": [
      "off"
    ],
    "no-constant-condition": [
      "warn"
    ],
    "no-constructor-return": [
      "error"
    ],
    "no-control-regex": [
      "off"
    ],
    "no-debugger": [
      "warn"
    ],
    "no-duplicate-case": [
      "error"
    ],
    "no-empty": [
      "off"
    ],
    "no-empty-pattern": [
      "off"
    ],
    "no-eval": [
      "warn"
    ],
    "no-fallthrough": [
      "off"
    ],
    "no-floating-decimal": [
      "warn"
    ],
    "no-implicit-coercion": [
      "warn"
    ],
    "no-inner-declarations": [
      "off"
    ],
    "no-invalid-this": [
      "off"
    ],
    "no-lonely-if": [
      "warn"
    ],
    "no-multiple-empty-lines": [
      "warn",
      {
        "max": 1,
        "maxBOF": 0,
        "maxEOF": 0
      }
    ],
    "no-multi-spaces": [
      "warn"
    ],
    "no-new-wrappers": [
      "warn"
    ],
    "no-octal-escape": [
      "warn"
    ],
    "no-regex-spaces": [
      "off"
    ],
    "no-return-await": [
      "warn"
    ],
    "no-sequences": [
      "warn"
    ],
    "no-shadow": [
      "off"
    ],
    "no-sparse-arrays": [
      "warn"
    ],
    "no-throw-literal": [
      "off"
    ],
    "no-trailing-spaces": [
      "warn"
    ],
    "no-undef-init": [
      "warn"
    ],
    "no-underscore-dangle": [
      "warn"
    ],
    "no-unsafe-finally": [
      "warn"
    ],
    "no-unused-labels": [
      "warn"
    ],
    "no-useless-catch": [
      "off"
    ],
    "no-useless-rename": [
      "warn"
    ],
    "no-var": [
      "warn"
    ],
    "no-void": [
      "warn"
    ],
    "no-whitespace-before-property": [
      "warn"
    ],
    "object-curly-newline": [
      "warn"
    ],
    "object-curly-spacing": [
      "warn",
      "always"
    ],
    "object-property-newline": [
      "warn",
      {
        "allowAllPropertiesOnSameLine": true,
        "allowMultiplePropertiesPerLine": false
      }
    ],
    "object-shorthand": [
      "warn"
    ],
    "one-var": [
      "warn",
      "never"
    ],
    "operator-linebreak": [
      "warn",
      "after"
    ],
    "padded-blocks": [
      "warn",
      "never"
    ],
    "padding-line-between-statements": [
      "warn",
      {
        "blankLine": "always",
        "prev": "*",
        "next": "multiline-block-like"
      },
      {
        "blankLine": "always",
        "prev": "multiline-block-like",
        "next": "*"
      },
      {
        "blankLine": "any",
        "prev": "multiline-block-like",
        "next": [
          "return",
          "break",
          "continue"
        ]
      }
    ],
    "prefer-const": [
      "off"
    ],
    "prefer-object-spread": [
      "warn"
    ],
    "prefer-rest-params": [
      "warn"
    ],
    "prefer-spread": [
      "warn"
    ],
    "quote-props": [
      "warn",
      "as-needed"
    ],
    "radix": [
      "warn"
    ],
    "rest-spread-spacing": [
      "warn"
    ],
    "semi-spacing": [
      "warn"
    ],
    "semi-style": [
      "warn"
    ],
    "space-before-blocks": [
      "warn"
    ],
    "space-in-parens": [
      "warn"
    ],
    "space-infix-ops": [
      "warn"
    ],
    "space-unary-ops": [
      "warn"
    ],
    "spaced-comment": [
      "warn"
    ],
    "switch-colon-spacing": [
      "warn"
    ],
    "template-curly-spacing": [
      "warn"
    ],
    "use-isnan": [
      "warn"
    ],
    "yield-star-spacing": [
      "warn"
    ],
    "yoda": [
      "warn"
    ],
    "@typescript-eslint/await-thenable": [
      "warn"
    ],
    "@typescript-eslint/no-throw-literal": [
      "warn"
    ],
    "@typescript-eslint/no-floating-promises": [
      "off"
    ],
    "@typescript-eslint/no-unnecessary-boolean-literal-compare": [
      "warn"
    ],
    "@typescript-eslint/no-unnecessary-qualifier": [
      "warn"
    ],
    "@typescript-eslint/no-unnecessary-type-arguments": [
      "warn"
    ],
    "@typescript-eslint/no-unnecessary-type-assertion": [
      "warn"
    ],
    "@typescript-eslint/prefer-includes": [
      "warn"
    ],
    "@typescript-eslint/prefer-nullish-coalescing": [
      "warn"
    ],
    "@typescript-eslint/prefer-string-starts-ends-with": [
      "warn"
    ],
    "@typescript-eslint/restrict-plus-operands": [
      "warn",
      {
        "checkCompoundAssignments": true
      }
    ],
    "@typescript-eslint/return-await": [
      "warn"
    ],
    "@typescript-eslint/unbound-method": [
      "warn"
    ],
    "@typescript-eslint/tslint/config": [
      "warn",
      {
        "lintFile": "E:/Users/Manuel/Documents/Projekte/Node-Modules/ESLintPresets/tslint.json"
      }
    ],
    "camelcase": [
      "off"
    ],
    "no-array-constructor": [
      "off"
    ],
    "@typescript-eslint/no-array-constructor": [
      "error"
    ],
    "no-empty-function": [
      "off"
    ],
    "@typescript-eslint/no-non-null-assertion": [
      "warn"
    ],
    "no-unused-vars": [
      "off"
    ],
    "no-use-before-define": [
      "off"
    ],
    "for-direction": [
      "error"
    ],
    "no-class-assign": [
      "error"
    ],
    "no-compare-neg-zero": [
      "error"
    ],
    "no-delete-var": [
      "error"
    ],
    "no-empty-character-class": [
      "error"
    ],
    "no-ex-assign": [
      "error"
    ],
    "no-extra-boolean-cast": [
      "error"
    ],
    "no-extra-semi": [
      "error"
    ],
    "no-func-assign": [
      "error"
    ],
    "no-global-assign": [
      "error"
    ],
    "no-invalid-regexp": [
      "error"
    ],
    "no-irregular-whitespace": [
      "error"
    ],
    "no-misleading-character-class": [
      "error"
    ],
    "no-mixed-spaces-and-tabs": [
      "error"
    ],
    "no-obj-calls": [
      "error"
    ],
    "no-octal": [
      "error"
    ],
    "no-prototype-builtins": [
      "error"
    ],
    "no-self-assign": [
      "error"
    ],
    "no-shadow-restricted-names": [
      "error"
    ],
    "no-unexpected-multiline": [
      "error"
    ],
    "no-unsafe-negation": [
      "error"
    ],
    "no-useless-escape": [
      "error"
    ],
    "no-with": [
      "error"
    ],
    "require-yield": [
      "error"
    ]
  },
  "settings": {},
  "ignorePatterns": []
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
switch (0)
{
    case 1:
        break;
    default:
        break;
    default:
        break;
}
```

```bash
eslint --max-warnings 0 --ext .js,.jsx,.ts,.tsx ./src
```

**What did you expect to happen?**
ESLint should report an error or a warning complaining about the duplicate `default`-case

**What actually happened? Please include the actual, raw output from ESLint.**
ESLint reported neither an error nor a warning.

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by mdjermanovic** [comment]
Hi @manuth, thanks for the bug report!

`no-duplicate-case` works as expected, this seems to be a problem with `@typescript-eslint/parser`.

Duplicate `default` should be parsing error, like in this [online demo](https://eslint.org/demo#eyJ0ZXh0Ijoic3dpdGNoICgwKVxue1xuICAgIGNhc2UgMTpcbiAgICAgICAgYnJlYWs7XG4gICAgZGVmYXVsdDpcbiAgICAgICAgYnJlYWs7XG4gICAgZGVmYXVsdDpcbiAgICAgICAgYnJlYWs7XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjoyLCJmb3ItZGlyZWN0aW9uIjoyLCJnZXR0ZXItcmV0dXJuIjoyLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjoyLCJuby1jYXNlLWRlY2xhcmF0aW9ucyI6Miwibm8tY2xhc3MtYXNzaWduIjoyLCJuby1jb21wYXJlLW5lZy16ZXJvIjoyLCJuby1jb25kLWFzc2lnbiI6Miwibm8tY29uc3QtYXNzaWduIjoyLCJuby1jb25zdGFudC1jb25kaXRpb24iOjIsIm5vLWNvbnRyb2wtcmVnZXgiOjIsIm5vLWRlYnVnZ2VyIjoyLCJuby1kZWxldGUtdmFyIjoyLCJuby1kdXBlLWFyZ3MiOjIsIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6Miwibm8tZHVwZS1rZXlzIjoyLCJuby1kdXBsaWNhdGUtY2FzZSI6Miwibm8tZW1wdHkiOjIsIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6Miwibm8tZW1wdHktcGF0dGVybiI6Miwibm8tZXgtYXNzaWduIjoyLCJuby1leHRyYS1ib29sZWFuLWNhc3QiOjIsIm5vLWV4dHJhLXNlbWkiOjIsIm5vLWZhbGx0aHJvdWdoIjoyLCJuby1mdW5jLWFzc2lnbiI6Miwibm8tZ2xvYmFsLWFzc2lnbiI6Miwibm8taW5uZXItZGVjbGFyYXRpb25zIjoyLCJuby1pbnZhbGlkLXJlZ2V4cCI6Miwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOjIsIm5vLW1pc2xlYWRpbmctY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1taXhlZC1zcGFjZXMtYW5kLXRhYnMiOjIsIm5vLW5ldy1zeW1ib2wiOjIsIm5vLW9iai1jYWxscyI6Miwibm8tb2N0YWwiOjIsIm5vLXByb3RvdHlwZS1idWlsdGlucyI6Miwibm8tcmVkZWNsYXJlIjoyLCJuby1yZWdleC1zcGFjZXMiOjIsIm5vLXNlbGYtYXNzaWduIjoyLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6Miwibm8tc3BhcnNlLWFycmF5cyI6Miwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOjIsIm5vLXVuZGVmIjoyLCJuby11bmV4cGVjdGVkLW11bHRpbGluZSI6Miwibm8tdW5yZWFjaGFibGUiOjIsIm5vLXVuc2FmZS1maW5hbGx5IjoyLCJuby11bnNhZmUtbmVnYXRpb24iOjIsIm5vLXVudXNlZC1sYWJlbHMiOjIsIm5vLXVudXNlZC12YXJzIjoyLCJuby11c2VsZXNzLWNhdGNoIjoyLCJuby11c2VsZXNzLWVzY2FwZSI6Miwibm8td2l0aCI6MiwicmVxdWlyZS15aWVsZCI6MiwidXNlLWlzbmFuIjoyLCJ2YWxpZC10eXBlb2YiOjJ9LCJlbnYiOnt9fX0=) with the default parser. Core rules are not designed to handle invalid code.

You can open an issue in [typescript-eslint/typescript-eslint](https://github.com/typescript-eslint/typescript-eslint) repository.

**Comment by manuth** [comment]
Thank you so much for your rapid answer ☺️
